scripts/globals/abilities/mijin_gakure.lua

```diff
@@ -25,9 +25,8 @@ function onUseAbility(player,target,ability)
     target:takeDamage(dmg, player, dsp.attackType.SPECIAL, dsp.damageType.ELEMENTAL)
 
     player:setLocalVar("MijinGakure", 1)
-    if (player:getMod(dsp.mod.MIJIN_NAGI) > 0) then
-        player:delStatusEffect(dsp.effect.RERAISE)
-        player:addStatusEffect(dsp.effect.RERAISE,1,0,3600)
+    if (player:getMod(dsp.mod.MIJIN_RERAISE) > 0) then
+        player:sendReraise(1)
```

Comment:
if you do the core right you don't need to force send the menu. also this would arrive at the player before they finished falling down.

Reply:
> 
> ```c++
> // Check if we have a form of reraise
> if (m_hasRaise > 0 || (getMod(Mod::MIJIN_NAGI) != 0 && GetLocalVar("MijinGakure") != 0))
> ```
> 
> ```c++
>         // Check for Nagi equipped (MOD) and Mijin Gakure was used.
>         if (GetLocalVar("MijinGakure") != 0 && getMod(Mod::MIJIN_NAGI) != 0)
>         {
>             actionTarget.animation = 511;
>             hpReturned = (uint16)(GetMaxHP());
>         }
>         else if (m_hasRaise == 1)
> ```

https://github.com/DarkstarProject/darkstar/pull/6262/files#diff-b3a74064477a12e9d4c37b8668ffad22R1468

Reply:
> if (m_hasRaise > 0 || (getMod(Mod::MIJIN_NAGI) != 0 && GetLocalVar("MijinGakure") != 0))

Prior to submitting the changes in 8bc76a5, I tried using your example at charentity.cpp:1468, and it does not generate a Raise event (after using Mijin with Nagi and this check, no raise menu pops). There's likely another place that generates the Raise event. (OnRaise?). 

I opted for the compromise of pushing a reraise event directly if Nagi was present and tested Mijin with Nagi and RR3, With Nagi and RR1, With Nagi and no RR, With RR and No Nagi, and without RR and without Nagi, all of which returned the anticipated correct results.

If you can point me to another place to generate the raise event without either pushing Reraise directly or pushing the reraise effect, I'd be curious to see where it resides and how to leverage it to accomplish the task.

Reply:
> Prior to submitting the changes in 8bc76a5, I tried using your example at charentity.cpp:1468, and it does not generate

And I don't know you've tried it till you tell me :) however, while my oversight in my sample code neglects the issue of needing call to onRaise to happen first, another place I pointed at just happens to cover this:
```cpp	
void CCharEntity::Die(duration _duration)
{
    m_deathSyncTime = server_clock::now() + death_update_frequency;
    PAI->ClearStateStack();
    PAI->Internal_Die(_duration);

    // reraise modifiers
    if (this->getMod(Mod::RERAISE_I) > 0)
        m_hasRaise = 1;

    if (this->getMod(Mod::RERAISE_II) > 0)
        m_hasRaise = 2;

    if (this->getMod(Mod::RERAISE_III) > 0)
        m_hasRaise = 3;

    // Check if we have MIJIN_RERAISE
    if (m_hasRaise == 0 && this->getMod(Mod::MIJIN_RERAISE) > 0)
        m_hasRaise = 1;

    CBattleEntity::Die();
}
```
The only requirement is the value be non zero before we hit the dirt, and I believe this would satisfy that.